lib/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {
+  Component,
+  ReactNode,
+  AllHTMLAttributes,
+  Fragment,
+} from 'react';
+import ThemeConsumer from '../ThemeConsumer/ThemeConsumer';
+import Box from '../Box/Box';
+import FieldLabel from '../FieldLabel/FieldLabel';
+import FieldMessage from '../FieldMessage/FieldMessage';
+import classnames from 'classnames';
+import styles from './TextArea.css.js';
+import Text from '../Text/Text';
+
+type NativeTextAreaProps = AllHTMLAttributes<HTMLTextAreaElement>;
+interface TextAreaProps {
+  id: NonNullable<NativeTextAreaProps['id']>;
+  value: string;
+  onChange: NonNullable<NativeTextAreaProps['onChange']>;
+  onBlur?: NativeTextAreaProps['onBlur'];
+  onFocus?: NativeTextAreaProps['onFocus'];
+  label?: string;
+  secondaryLabel?: ReactNode;
+  tertiaryLabel?: ReactNode;
+  placeholder?: string;
+  message?: ReactNode | false;
+  tone?: 'neutral' | 'critical' | 'positive';
+  description?: 'string';
+  limit?: number;
+}
+
+export default class TextArea extends Component<TextAreaProps> {
+  static displayName = 'TextArea';
+
+  renderCount({
+    limit,
+    value = '',
+  }: Pick<TextAreaProps, 'limit' | 'value'>): ReactNode {
+    if (typeof limit === 'undefined' || typeof value === 'undefined') {
```

Comment:
Just realised, since we're defaulting value to a blank string, it can never be undefined.